### PR TITLE
gh-102304: doc: Add links to Stable ABI and Limited C API

### PR DESCRIPTION
**Pull request by vstinner**
* Add "limited-c-api" and "stable-api" references.
* Rename "stable-abi-list" reference to "limited-api-list".
* Makefile: Document files regenerated by "make regen-limited-abi"
* Remove first empty line in generated files:

  - Lib/test/test_stable_abi_ctypes.py
  - PC/python3dll.c

* Issue: gh-102304

----
:books: Documentation preview :books:: https://cpython-previews--105345.org.readthedocs.build/

### CODE REVIEW

**Comment by encukou** [comment]
As mentioned on the other PR, this is wrong. Please take my review comment into account.

File: Makefile.pre.in [hunk 1]
```diff
@@ -1309,6 +1309,11 @@ check-abidump: all
 
 .PHONY: regen-limited-abi
 regen-limited-abi: all
+	# Regenerate files using using Tools/build/stable_abi.py:
+	# - Doc/data/stable_abi.dat
+	# - Lib/test/test_stable_abi_ctypes.py
+	# - Modules/_testcapi_feature_macros.inc
+	# - PC/python3dll.c
```
Comment (encukou) [inline Makefile.pre.in:1316]: Do you volunteer to keep this list up to date?

File: Doc/c-api/stable.rst [hunk 2]
```diff
@@ -87,6 +83,23 @@ embedding Python.)
...
+To enable this, Python provides a *Stable ABI*: a set of symbols that will
+remain compatible across Python 3.x versions.
+
+The Stable ABI contains symbols exposed in the :ref:`Limited API
+<limited-c-api>`, but also other ones – for example, functions necessary to
+support older versions of the Limited API.
+
+(For simplicity, this document talks about *extensions*, but the Limited API
+and Stable ABI work the same way for all uses of the API – for example,
+embedding Python.)
```
Comment (encukou) [inline Doc/c-api/stable.rst:101]: This applies to the previous section as well. could you move it up under 
"Stable Application Binary Interface"?

**Comment by encukou** [comment]
A few nitpicks, but this looks great overall! Thank you!

File: Makefile.pre.in (hunk 1 above)
Comment (vstinner) [reply Makefile.pre.in:1316]: It's ok if it's outdated, it's just a comment.

File: Doc/c-api/stable.rst (hunk 2 above)
Comment (vstinner) [reply Doc/c-api/stable.rst:101]: Done in PR #105351.

File: Makefile.pre.in (hunk 1 above)
Comment (encukou) [reply Makefile.pre.in:1316]: If it's not necessary, please remove it. You can say how to get the up-to date info, but I don't think that's needed either.

Comment (vstinner) [reply Makefile.pre.in:1316]: I added the URL that you proposed in PR #105351.

**Comment by AA-Turner** [comment]
Looking to backport this to 3.11 for https://github.com/python/devguide/pull/1167.

A

**Pull request merged**